### PR TITLE
Add better description and guidance in case of sqlite version mismatch

### LINKED ISSUE #14208: Python 3.8 - Sqlite3 version error
python 3.8 centos 7 Docker image

Can't update sqlite3.  Tried Airflow 2.0.1 and 2.0.0.  Same issue on Python 3.6 with Airflow 2.0.0.  I was able to force the install on 2.0.0 but when running a task it failed because of the sqlite3 version mismatch.  

Am I just stupid?  #13496 

 #```
(app-root) airflow db init
Traceback (most recent call last):
  File "/opt/app-root/bin/airflow", line 5, in <module>
    from airflow.__main__ import main
  File "/opt/app-root/lib64/python3.8/site-packages/airflow/__init__.py", line 34, in <module>
    from airflow import settings
  File "/opt/app-root/lib64/python3.8/site-packages/airflow/settings.py", line 37, in <module>
    from airflow.configuration import AIRFLOW_HOME, WEBSERVER_CONFIG, conf  # NOQA F401
  File "/opt/app-root/lib64/python3.8/site-packages/airflow/configuration.py", line 1007, in <module>
    conf.validate()
  File "/opt/app-root/lib64/python3.8/site-packages/airflow/configuration.py", line 209, in validate
    self._validate_config_dependencies()
  File "/opt/app-root/lib64/python3.8/site-packages/airflow/configuration.py", line 246, in _validate_config_dependencies
    raise AirflowConfigException(f"error: cannot use sqlite version < {min_sqlite_version}")
airflow.exceptions.AirflowConfigException: error: cannot use sqlite version < 3.15.0
(app-root) python -c "import sqlite3; print(sqlite3.sqlite_version)"
3.7.17
(app-root) python --version
Python 3.8.6
(app-root) pip install --upgrade sqlite3
ERROR: Could not find a version that satisfies the requirement sqlite3 (from versions: none)
ERROR: No matching distribution found for sqlite3
```

### PR DESCRIPTION
**Pull request by potiuk**
Closes: #14208

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/configuration.py [hunk 1]
```diff
@@ -243,7 +243,11 @@ def _validate_config_dependencies(self):
             # Some of the features in storing rendered fields require sqlite version >= 3.15.0
             min_sqlite_version = '3.15.0'
             if StrictVersion(sqlite3.sqlite_version) < StrictVersion(min_sqlite_version):
-                raise AirflowConfigException(f"error: cannot use sqlite version < {min_sqlite_version}")
+                raise AirflowConfigException(
+                    f"error: cannot use sqlite version < {min_sqlite_version}. "
+                    f"See https://github.com/apache/airflow/blob/master/docs/apache-airflow/"
+                    f"howto/set-up-database.rst#sqlite-database-setup"
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
                    f"See {docs_url("howto/set-up-database.rst#sqlite-database-setup")}"
```

(`from airflow.utils.docs import docs_url`)

File: airflow/configuration.py [hunk 2]
```diff
@@ -243,7 +243,11 @@ def _validate_config_dependencies(self):
             # Some of the features in storing rendered fields require sqlite version >= 3.15.0
             min_sqlite_version = '3.15.0'
             if StrictVersion(sqlite3.sqlite_version) < StrictVersion(min_sqlite_version):
-                raise AirflowConfigException(f"error: cannot use sqlite version < {min_sqlite_version}")
+                raise AirflowConfigException(
+                    f"error: cannot use sqlite version < {min_sqlite_version}. "
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
                    f"error: sqlite C library version too old (< {min_sqlite_version}). "
```

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by potiuk** [comment]
Updated @ashb

**Review by ashb — APPROVED** [review]

File: docs/apache-airflow/howto/set-up-database.rst [hunk 3]
```diff
@@ -59,6 +59,45 @@ the example below.
 
 The exact format description is described in the SQLAlchemy documentation, see `Database Urls <https://docs.sqlalchemy.org/en/14/core/engines.html>`__. We will also show you some examples below.
 
+SQLite database setup
+---------------------
+
+SQLite database can be used to run Airflow for development purpose as it does not require any database server
```
Comment (mik-laj) [inline docs/apache-airflow/howto/set-up-database.rst:65]: Can you add docs about connection URI?  The sections for other databases cover at least one example of such a URI, so adding an example in this section would be helpful.

Comment (potiuk) [reply docs/apache-airflow/howto/set-up-database.rst:65]: Good point. Added.

**Review by mik-laj — APPROVED** [review]

File: docs/apache-airflow/howto/set-up-database.rst [hunk 4]
```diff
@@ -59,6 +59,51 @@ the example below.
 
 The exact format description is described in the SQLAlchemy documentation, see `Database Urls <https://docs.sqlalchemy.org/en/14/core/engines.html>`__. We will also show you some examples below.
 
+SQLite database setup
+---------------------
```
Comment (mik-laj) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Setting up a SQLite Database
----------------------------
```
To keep one style of headings.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**